airbyte-integrations/connectors/destination-bigquery-denormalized/src/test-integration/resources/testdata/expectedDataArrays.json

```diff
@@ -0,0 +1,33 @@
+{
+  "object_with_arrays": {
+    "array_3": [1, 2, 3]
+  },
+  "simple_string": "simple string",
+  "array_1": [
+    {
+      "big_query_array": [1]
+    }
+  ],
+  "array_4": [
+    {
+      "big_query_array": [
+        {
+          "big_query_array": [4]
+        }
+      ]
+    }
+  ],
+  "array_5": [
+    {
+      "big_query_array": [
+        {
+          "big_query_array": [
+            {
+              "big_query_array": [5]
```

Comment:
then, we should have a test case with 15 levels and a test case with 16 levels to be sure that the connector behaves as expected for edge cases.

Reply:
I didn't cover these cases because it is not a limitation or specific to our implementation. Our implementation doesn't have any depth limit. So, the test of 3 nested arrays proves the algorithm's scalability is the same as for 15.
On the other hand, the test of 16 levels might be unstable because the BQ can change it at any moment, and our test will fail.

Reply:
> I didn't cover these cases because it is not a limitation or specific to our implementation. Our implementation doesn't have any depth limit. So, the test of 3 nested arrays proves the algorithm's scalability is the same as for 15. On the other hand, the test of 16 levels might be unstable because the BQ can change it at any moment, and our test will fail.

here's what I would like to confirm:
 - our connector can handle 15 levels of nesting
 - when source data has more levels of nesting than destination can support - we handle the error correctly and behavior is deterministic

Reply:
Follow-up issue #14668